NDTensors/src/zeros/zeros.jl

```diff
@@ -0,0 +1,35 @@
+struct Zeros{ElT,N,DataT} <: AbstractArray{ElT,N}
+  z::FillArrays.Zeros
+  is::Tuple
+  function NDTensors.Zeros{ElT,N,DataT}(inds::Tuple) where {ElT,N,DataT}
+    @assert eltype(DataT) == ElT
+    @assert ndims(DataT) == N
+    z = FillArrays.Zeros(ElT, dim(inds))
+    return new{ElT,N,DataT}(z, inds)
+  end
+end
+
+Base.ndims(::NDTensors.Zeros{ElT,N}) where {ElT,N} = N
+ndims(::NDTensors.Zeros{ElT,N}) where {ElT,N} = N
+Base.eltype(::Zeros{ElT}) where {ElT} = ElT
+datatype(::NDTensors.Zeros{ElT,N,DataT}) where {ElT,N,DataT} = DataT
```

Comment:
Maybe a better name would be something like `allocated_type`, `datatype` is pretty generic and has a different meaning in the rest of the library.